src/metabase/lib/field.cljc

```diff
@@ -378,7 +385,7 @@
 (defmethod lib.temporal-bucket/available-temporal-buckets-method :metadata/column
   [_query _stage-number field-metadata]
   (if (not= (:lib/source field-metadata) :source/expressions)
-    (let [effective-type ((some-fn :effective-type :base-type) field-metadata)
+    (let [effective-type ((some-fn ::original-effective-type :effective-type :base-type) field-metadata)
```

Comment:
`with-temporal-bucket` + `available-temporal-buckets` is already tested for extraction units. When I fixed `effective-type` for such columns this place broke, so I had to fix it.